shared/AppInsightsCore/Tests/Unit/src/CookieManager.Tests.ts

```diff
@@ -749,6 +774,513 @@ export class CookieManagerTests extends AITestClass {
+                Assert.equal(newValue + "; path=/", this._testCookies[newKey], "Cookie should be flushed to actual storage");
+                
+                // Get should still return the value
+                Assert.equal(newValue, manager.get(newKey), "Should return value from actual storage");
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Write cookies after being enabled - multiple cookies",
+            test: () => {
+                let manager = createCookieMgr(this._config);
+
+                // Start with cookies disabled
+                manager.setEnabled(false);
+
+                let cookies = [
+                    { key: "test1." + newId(), value: newId() },
+                    { key: "test2." + newId(), value: newId() },
+                    { key: "test3." + newId(), value: newId() }
+                ];
+                
+                // Set multiple cookies while disabled
+                cookies.forEach(cookie => {
+                    let result = manager.set(cookie.key, cookie.value);
+                    Assert.equal(true, result, "Set should return true even when disabled");
+                    Assert.equal(cookie.value, manager.get(cookie.key), "Should return cached value");
+                    Assert.equal(undefined, this._testCookies[cookie.key], "Cookie should not be in actual storage yet");
+                });
+
+                // Enable cookies - should flush all cached values
+                manager.setEnabled(true);
+                
+                // All cookies should now be in actual storage
+                cookies.forEach(cookie => {
+                    Assert.equal(cookie.value + "; path=/", this._testCookies[cookie.key], "Cookie should be flushed to actual storage");
+                    Assert.equal(cookie.value, manager.get(cookie.key), "Should return value from actual storage");
+                });
+            }
+        });
+
+        this.testCase({
+            name: "CookieManager: Write cookies after being enabled - with maxAge and domain",
+            test: () => {
+                this._cookieMgrCfg.domain = "test.com";
+                let manager = createCookieMgr(this._config);
+
+                // Start with cookies disabled
+                manager.setEnabled(false);
+
+                let newKey = "test." + newId();
+                let newValue = newId();
+                let maxAge = 3600; // 1 hour
+                
+                // Set a cookie with maxAge while disabled
+                let result = manager.set(newKey, newValue, maxAge);
+                Assert.equal(true, result, "Set should return true even when disabled");
+                Assert.equal(newValue, manager.get(newKey), "Should return cached value");
+
+                // Enable cookies
+                manager.setEnabled(true);
+                
+                // Cookie should be flushed with the maxAge parameter
```

Comment:
You need to add a clock.tick(1) after changing the state to allow the dynamic configurations changes to take affect